### PR TITLE
Fix EZP-24383: Implement form action dispatcher

### PR DESCRIPTION
**Pull request by lolautruche**
> https://jira.ez.no/browse/EZP-24383

This PR introduces a simple mechanism to handle several actions in a single form.
Every time a form using the mechanism is submitted (and validated), a `FormActionEvent` is fired. This event contains the form, its underlying data, the clicked button name and the language code.

In the case of ContentType, event names are `contentType.update.<buttonName>`, which makes it very easy to attach a specific *form processor* (which is actually an event subscriber).

`FormDataMapperInterface` is also introduced to help converting repository value objects to the underlying form data. This will help avoiding doing this in the controller.

### CODE REVIEW

**Comment by lolautruche** [comment]
Review ping @andrerom @bdunogier @dpobel @yannickroger @pspanja @glye

File: bundle/Controller/ContentTypeController.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+use eZ\Bundle\EzPublishCoreBundle\Controller;
+use eZ\Publish\Core\Repository\Values\ContentType\ContentTypeCreateStruct;
+use EzSystems\RepositoryForms\Data\Mapper\ContentTypeDraftMapper;
+use EzSystems\RepositoryForms\Event\FormActionEvent;
+use EzSystems\RepositoryForms\Event\RepositoryFormEvents;
+use Symfony\Component\HttpFoundation\Request;
+
+class ContentTypeController extends Controller
+{
+    public function createContentTypeAction($contentTypeGroupId, $languageCode = null)
```
Comment (bdunogier) [inline bundle/Controller/ContentTypeController.php]: Could we use the ParamConverter thing for the ContentTypeGroup ?

**Comment by glye** [comment]
+1

File: bundle/Controller/ContentTypeController.php (hunk 1 above)
Comment (lolautruche) [inline bundle/Controller/ContentTypeController.php]: OutOfScopeException :smile: 

Follow-up story created: https://jira.ez.no/browse/EZP-24409

File: bundle/Controller/ContentTypeController.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        return $this->redirectToRoute(
+            'contenttype/update',
+            ['contentTypeId' => $contentTypeDraft->id, 'languageCode' => $languageCode]
+        );
+    }
+
+    public function updateContentTypeAction(Request $request, $contentTypeId, $languageCode = null)
+    {
+        $languageCode = $languageCode ?: $this->getConfigResolver()->getParameter('languages')[0];
```
Comment (bdunogier) [inline bundle/Controller/ContentTypeController.php]: No controller as a service, and params/services injection ?

**Comment by bdunogier** [comment]
+1 besides the remarks

File: bundle/Controller/ContentTypeController.php (hunk 2 above)
Comment (lolautruche) [inline bundle/Controller/ContentTypeController.php]: Well, we'd have a lot of dependencies for very low gain IMHO.
Services needed:
* Templating
* Router
* FormFactory
* EventDispatcher
* ContentTypeService
* FieldTypeFormMapperRegistry

+ of course language list as injected setting.

**Comment by lolautruche** [comment]
I tried to the controller as a service and finally ended up to rewrite most helper methods from `\Symfony\Bundle\FrameworkBundle\Controller\Controller`. So doesn't sound like a good fit here.

**Pull request merged**